frontend/gettext.lua

```diff
@@ -57,20 +145,43 @@ function GetText_mt.__index.changeLang(new_lang)
     end
 
     local data = {}
+    local headers
     local what = nil
     while true do
         local line = po:read("*l")
         if line == nil or line == "" then
-            if data.msgid and data.msgstr and data.msgstr ~= "" then
-                local unescaped_string = string.gsub(data.msgstr, "\\(.)", c_escape)
-                if data.msgctxt and data.msgctxt ~= "" then
-                    if not GetText.context[data.msgctxt] then
-                        GetText.context[data.msgctxt] = {}
+            if data.msgid and data.msgid_plural and data["msgstr[0]"] then
+                for k, v in pairs(data) do
+                    local n = tonumber(k:match("msgstr%[([0-9]+)%]"))
+                    local msgstr = v
+
+                    if n and msgstr then
+                        addTranslation(data.msgctxt, data.msgid, msgstr, n)
                     end
-                    GetText.context[data.msgctxt][data.msgid] = unescaped_string
-                else
-                    GetText.translation[data.msgid] = unescaped_string
                 end
+            elseif data.msgid and data.msgstr and data.msgstr ~= "" then
+                -- header
+                if not headers and data.msgid == "" then
+                    local util = require("util")
+                    headers = data.msgstr
+                    local plural_forms = data.msgstr:match("Plural%-Forms: (.*);")
+                    local nplurals = plural_forms:match("nplurals=([0-9]+);") or 1
```

Comment:
2 actually seems to be right; I'll have to look at the docs a bit better. Clearly you don't have to declare the default.

Reply:
Oh, I guess perhaps I can't check for sanity because a C boolean expression is 0 or 1.